### PR TITLE
added docs for all Field options

### PR DESCRIPTION
**Pull request by paddymul**

### CODE REVIEW

**Comment by paddymul** [comment]
@Arsik36 @hansjhoffman  Take a look at this Hans, can you look at the stage visibility explanation?

File: README.md [hunk 1]
```diff
@@ -459,6 +459,30 @@ The default value for this field
 `validate` takes the type specified by the field and returns validation messages.  This is the most commonly used field hook.
 
 
+#### Other Field Options
+##### Description
+Long form description that appears in the UI
```
Comment (hansjhoffman) [inline README.md:464]: ⛳ "Long form description that appears in the UI upon hover of the field name."

File: README.md [hunk 2]
```diff
@@ -459,6 +459,30 @@ The default value for this field
...
+##### Annotations
+Annotations are automatically filled in messages that the platform sdk provides when `default` or `compute` changes a value. Following indepent optyions set as an object - `{}`
```
Comment (hansjhoffman) [inline README.md:466]: 🔧  Typo in "options" here with "optyions"

File: README.md [hunk 3]
```diff
@@ -459,6 +459,30 @@ The default value for this field
...
+###### default
+If set to `true` insertions of the `default` value will be annotated with an info message of `defaultMessage`
+###### defaultMessage
+The message to use when a `default` value is inserted.  if none specified, defaults to 'This field was automatically given a default value of'
```
Comment (hansjhoffman) [inline README.md:470]: 🔧 Capitalize "if"?

File: README.md [hunk 4]
```diff
@@ -459,6 +459,30 @@ The default value for this field
...
+###### compute
+If set to `true` instances where `compute` changes the value of a field  will be annotated with an info message of `computeMessage`
```
Comment (hansjhoffman) [inline README.md:472]: ⛳ Looks like 2 spaces between "field will"

File: README.md [hunk 5]
```diff
@@ -459,6 +459,30 @@ The default value for this field
...
+###### computeMessage
+The message to use when a `compute` changes a value.  if none specified, defaults to 'This value was automatically reformatted - original data:'
+##### stageVisibility
+controls what parts of mapping/review/export a field occurs in
+###### mapping
+When set to `false` this field will not appear for matching in the mapping phase.
+###### review
+This field will not appear in the review stage
+###### export
+This field will not be exported.
```
Comment (hansjhoffman) [inline README.md:482]: ⛳ May want to use the same vernacular here. "phase" or "stage"

**Review by hansjhoffman — APPROVED** [review]

File: README.md (hunk 5 above)
Comment (paddymul) [reply README.md:482]: both are stage

File: README.md (hunk 4 above)
Comment (paddymul) [reply README.md:472]: Done

File: README.md (hunk 3 above)
Comment (paddymul) [reply README.md:470]: Done

**Pull request merged**